backend/cmm_helpers.ml

```diff
@@ -3646,18 +3679,19 @@ let preallocate_block cont { Clambda.symbol; exported; tag; fields } =
         match field with
         | None -> Cint (Nativeint.of_int 1 (* Val_unit *))
         | Some (Clambda.Uconst_field_int n) -> cint_const n
-        | Some (Clambda.Uconst_field_ref label) -> Csymbol_address label)
+        | Some (Clambda.Uconst_field_ref label) -> Csymbol_address (mksym label))
       fields
   in
-  let global = Cmmgen_state.(if exported then Global else Local) in
-  let symb = symbol, global in
-  let data = emit_block symb (block_header tag (List.length fields)) space in
+  let data =
+    emit_block (mksym symbol) (block_header tag (List.length fields)) space
+  in
   Cdata data :: cont
 
 let emit_preallocated_blocks preallocated_blocks cont =
   let symbols =
     List.map
-      (fun ({ Clambda.symbol } : Clambda.preallocated_block) -> symbol)
+      (fun ({ Clambda.symbol } : Clambda.preallocated_block) ->
+        global_symbol symbol)
```

Comment:
why does this code not use the `exported` field of a `Clambda.preallocated_block` to determine the locality ? Especially since the `preallocate_block` function above does use it.

Reply:
Well, it's not redundant because the `symbol` from a `preallocated_block` is a `string`. But, indeed the pair `symbol, exported` of a preallocated block should essentially be equivalent to a `Cmm.symbol`